packages/dd-trace/test/heap_snapshots.spec.js

```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+require('./setup/tap')
+
+const { mkdtempSync, readdirSync } = require('fs')
+const { tmpdir } = require('os')
+const { join } = require('path')
+const { start } = require('../src/heap_snapshots')
+
+const folder = mkdtempSync(join(tmpdir(), 'dd-trace-heap-snapshot-'))
+
+describe('Heap Snapshots', () => {
+  it('should take heap snapshots over time', async () => {
+    // Keep process alive since `start` uses an unref timer.
+    const interval = setInterval(() => {}, 1000)
+
+    await start({
+      heapSnapshot: {
+        count: 2,
+        folder,
+        interval: 1
+      }
+    })
+
+    clearInterval(interval)
+
+    const pattern = /^Heap-\d{8}-\d{6}-\d+-\d+\.heapsnapshot$/
```

Comment:
Not a big deal, but just a slight improvement to the validation now that you _are_ generating the filename manually (where `threadId` is imported at the top of the spec file). This would also catch cases where somehow files are reused between runs for some reason

```suggestion
    const pattern = new RegExp(`^Heap-\\d{8}-\\d{6}-${process.pid}-${threadId}\\.heapsnapshot$`)
```